app/controllers/csv_exports_controller.rb

```diff
@@ -0,0 +1,102 @@
+class CsvExportsController < ApplicationController
+
+  rescue_from ActiveRecord::RecordNotFound do |exception|
+    respond_to do |format|
+      format.json { render json: {status: "not found"}.to_json, status: :not_found }
+      format.csv  { render body: "not found", status: :not_found }
+      format.html { raise exception }
+    end
+  end
+
+  def index
+    @csv_exports = CsvExport.where(user_id: current_user.id)
+    respond_to do |format|
+      format.html
+      format.json { render json: @csv_exports }
+    end
+  end
+
+  def new
+    @csv_export = CsvExport.new
+  end
+
+  def show
+    @csv_export = CsvExport.find(params[:id])
+    respond_to do |format|
+      format.html
+      format.json { render json: @csv_export }
+      format.csv { download }
+    end
+  end
+
+  def create
+    filters = deploy_filter(params)
+    csv_export = CsvExport.create!(user: current_user, filters: filters)
+    CsvExportJob.perform_later(csv_export)
+    redirect_to csv_export
+  end
+
+  private
+
+  def download
+    if @csv_export.status? :ready
+      begin
+        send_file @csv_export.path_file, type: :csv, filename: @csv_export.download_name
+        @csv_export.status! :downloaded
+        Rails.logger.info("#{current_user.name_and_email} just downloaded #{@csv_export.download_name})")
+      rescue ActionController::MissingFile
+        @csv_export.status! :deleted
+        redirect_to @csv_export
+      end
+    else
+      redirect_to @csv_export
+    end
+  end
+
+  def deploy_filter(params)
```

Comment:
you shouldn't pass in params, it will already be available in private controller methods just as it is in an action